MatrixSDKTests/Utils/MXTaskQueueUnitTests.swift

```diff
@@ -301,18 +310,33 @@ class MXTaskQueueUnitTests: XCTestCase {
         }
     }
     
+    /// Assert that for every task id all operations (task start, task end and non task) are performed
     private func XCTAssertAllOperationsPerformed(_ taskIds: [String], file: StaticString = #file, line: UInt = #line) async {
         let count = await timeline.numberOfOperations
         XCTAssertEqual(count, taskIds.count * Operation.Kind.allCases.count, file: file, line: line)
     }
     
-    private func XCTAssertTaskOrderEqual(_ taskIds: [String], expectedOrder: [Operation.Kind], file: StaticString = #file, line: UInt = #line) async {
+    /// Assert that operations for each task happen in the exact order specified
+    private func XCTAssertTaskOrderEquals(_ taskIds: [String], expectedOrder: [Operation.Kind], file: StaticString = #file, line: UInt = #line) async {
         for id in taskIds {
             let realOrder = await timeline.operationOrder(for: id)
             XCTAssertEqual(realOrder, expectedOrder, "Order for task \(id) is incorrect", file: file, line: line)
         }
     }
     
+    /// Assert that for every task a given operation occurs before another operation
+    private func XCTAssertOperationBefore(_ taskIds: [String], operation: Operation.Kind, before: Operation.Kind, file: StaticString = #file, line: UInt = #line) async {
+        for id in taskIds {
+            let realOrder = await timeline.operationOrder(for: id)
+            guard let index = realOrder.firstIndex(of: operation), let beforeIndex = realOrder.firstIndex(of: before) else {
+                XCTFail("Cannot find given operations", file: file, line: line)
+                return
+            }
+            XCTAssertLessThan(index, beforeIndex, "Operation \(operation) does not happen before \(before)", file: file, line: line)
```

Comment:
Naming wise, `beforeIndex` should come after `index` is a bit hard to parse. Maybe more helpful to define something like `first`/`secondOperation` and `first`/`secondIndex` or similar.